src/status_im/subs/navigation.cljs

```diff
@@ -0,0 +1,11 @@
+(ns status-im.subs.navigation
+  (:require
+    [re-frame.core :as re-frame]))
+
+(re-frame/reg-sub
+ :navigation/current-screen-id
+ :<- [:view-id]
+ :<- [:modal-view-ids]
+ (fn [[view-id modal-view-ids]]
+   (let [modal-view-ids (not-empty modal-view-ids)]
+     (if modal-view-ids (last modal-view-ids) view-id))))
```

Comment:
If `modal-view-ids` is a vector we can use `peek` (fast `last` for vectors) and it works with `nil`, so we don't need to check if it's `not-empty` or not.
so

```clojure
(fn [[view-id modal-view-ids]]
   (or (peek modal-view-ids) view-id))
```
should do the job, substitute `peek` for `last` if we are dealing with list or the type is unsure.

Reply:
Nice one, thanks for the suggestion!